lib/evaluate/intrinsics-library.h

```diff
@@ -35,7 +35,7 @@ class FoldingContext;
 using TypeCode = unsigned char;
 
 template<typename TR, typename... TA> using FuncPointer = TR (*)(TA...);
-using GenericFunctionPointer = FuncPointer<void *>;
+using GenericFunctionPointer = void (*) (void);
```

Comment:
Please add a comment to the effect that this exact type has magical properties for GCC.